### DISCUSSION TITLE
Custom rule support

### DISCUSSION THREAD

**Discussion by aliavni** [astral-sh/ruff#8409] [Q&A] [answered]
Hi, is it possible to add custom rules to ruff? The idea is to add a repo specific rule that is not covered by the common rules ruff supports. For example not allowing `getattr`/`setattr` calls.

**Comment by tjkuson** [comment, marked as answer]
If you mean plugins, there is no plugin system yet (though there may be in the future). There is an issue that discusses the pros and cons, and how such a system could be implemented: #283.

**Reply by aliavni** [reply]
@tjkuson yes I had plugins in mind. I think it would be super useful. I will follow the discussion you linked. Thanks a lot!
